src/layer.ts

```diff
@@ -74,17 +74,33 @@ export function applyLayers(region: string, handlers: FunctionInfo[], layers: La
     }
 
     const { runtime } = handler;
-    const layerARN = runtime !== undefined ? regionRuntimes[runtime] : undefined;
-    if (layerARN !== undefined) {
-      const currentLayers = getLayers(handler);
-      if (!new Set(currentLayers).has(layerARN)) {
-        currentLayers.push(layerARN);
-      }
+    const lambdaLayerARN = runtime !== undefined ? regionRuntimes[runtime] : undefined;
+    let extensionLayerARN: string | undefined;
+    if (applyExtensionLayer) extensionLayerARN = regionRuntimes[extensionLayerKey];
+    let currentLayers = getLayers(handler);
+
+    if (lambdaLayerARN && extensionLayerARN) {
```

Comment:
Do you need three cases here? I.e. do you need to check `lambdaLayerARN && extensionLayerARN`, `lambdaLayerARN`, and `extensionLayerARN` separately? Can you call `setLayers` more than once? If so, then I think you might only need to check for the `lambdaLayerARN` and `extensionLayerARN` separately.

Reply:
I refactored the logic